LDAR-Sim_inputs.md

```diff
@@ -26,6 +26,263 @@ To support wider use of LDAR-Sim, the University of Calgary and Highwood Emissio
+
+```buildoutcfg
+new_LDAR_method_1 = {
+    'version': '2.0',
+    'parameter_level': 'method',
+    'label': 'new_LDAR_method_1',
+    'module': 'OGI',
+    .... OTHER OGI METHOD PARAMETERS ....
+}
+```
+
+When the simulation is put together, the program will be assembled to look like this:
+
+```buildoutcfg
+test_program = {
+    'version': '2.0',
+    'parameter_level': 'program',
+    'method_names': ['new_LDAR_method_1'],
+    'methods': [
+        'new_LDAR_method_1': {
+            'parameter_level': 'method',
+            'label': 'OGI',
+            .... OTHER OGI METHOD PARAMETERS ....
+        }
+    ],
+    .... OTHER PROGRAM PARAMETERS ....
+}
+```
+
+To review, the following parameters are necessary to enable this modularization and reproducibility within the parameter suites:
+
+`parameter_level`: `global`, `program`, or `method`, this defines the target level in the hierarchy. Without specification, LDAR-Sim interprets the parameters as global.
+
+`label`: in method definitions `type` provides a lookup for checking parameters and allows lookup of default parameters, which enables partial parameter specification and out of the box reverse compatibility.
+
+`method_labels`: shorthand method to specify methods by their labels, and include them in more than one simulation easily and reliably. Used only in programs.
+
+## Parameter File Formats
+
+LDAR-Sim includes a flexible input parameter mapper that accepts a variety of input parameter formats, choose the one that you like the best. Yaml[YAML](https://en.wikipedia.org/wiki/YAML) is the easiest to read for humans, allows inline comments, and is recommended. The following formats are accepted:
+
+- yaml files (extension = '.yaml' or '.yml')
+- json files (extension = '.json')
+
+Keep in mind the need for precision with input parameter types and review how each of these formats handle input data types. For example, `10` is inferred as an integer, where `10.0` is inferred as a floating point value - these are different data types and will fail validation.
+
+For example, here is a program definition in yaml:
+
+```buildoutcfg
+version: '2.0'
+parameter_level: program
+```
+
+Here is a method definition in yaml:
+
+```buildoutcfg
+version: '2.0'  
+parameter_level: method
+awesome_method:
+  label: awesome_method
+  module: awesome_technology
+```
+
+Note that programs are interpreted as a flat list of parameters that are incorporated into a list where methods have one parameter (the method name), and other method parameters nested below.
+
+## Versioning of Parameter Files
+
+All parameter files must specify a version to enable mapping and reverse compatiblity. This versioning is used to call code that modifies a different version of the code to run properly. In cases this is simple mapping of parameters, in other cases, this involves calculations. Refer to `input_mapper_v1.py` for a template file and discussion document on input parameter mapping.
```

Comment:
Do we want to mention that reverse compatability only exists for v2 onwawrds?

Reply:
Added, thanks.